packages/core-plugin-api/api-report.md

```diff
@@ -149,9 +149,37 @@ export type AppComponents = {
 export type AppContext = {
   getPlugins(): BackstagePlugin_2[];
   getSystemIcon(key: string): IconComponent_2 | undefined;
+  getSystemIcons(): AppIcons & {
+    [key in string]: IconComponent_2;
+  };
   getComponents(): AppComponents;
 };
 
+// @public
+export type AppIcons = {
```

Comment:
I think it's best to skip having `AppIcons` and just have the return type be `Record<string, IconComponent>` instead, it's what we ended up doing for `getSystemIcon`.

Reply:
OK, cool, pulled this out and updated to use `Record<string, IconComponent>` as suggested 👍